### ISSUE TITLE
MOODS-python-1.9.1 error

### ISSUE THREAD

**Issue by ekaasine** [jhkorhonen/MOODS#6]
Hi Janne,

I installed the new MOODS-python-1.9.1 version successfully but when trying to run it using e.g. command "python moods_dna.py --help" in the scripts folder I get the following error:

Traceback (most recent call last):
  File "moods_dna.py", line 6, in <module>
    import MOODS.parsers
  File "/usr/local/lib/python2.7/dist-packages/MOODS/parsers.py", line 28, in <module>
    _parsers = swig_import_helper()
  File "/usr/local/lib/python2.7/dist-packages/MOODS/parsers.py", line 24, in swig_import_helper
    _mod = imp.load_module('_parsers', fp, pathname, description)
ImportError: /usr/local/lib/python2.7/dist-packages/MOODS/_parsers.so: undefined symbol: _ZN5MOODS7variantC1EmmSs

Any idea what's the problem? Thanks a lot!

Best,
Eevi

**Comment by jhkorhonen** [comment]
Hi Eevi,

it looks like the build script does not include all necessary sources when building `MOODS.parsers`, leading to missing symbols. This is a bug; I did not caught it in testing, as for things work out for me when loading `MOODS.tools` before `MOODS.parsers`.

Could you download the updated setup.py from [here](https://raw.githubusercontent.com/jhkorhonen/MOODS/develop/python/setup.py), replace the setup.py from 1.9.1 package with that one, re-build MOODS and verify that it works? I'll then update the package to include the fixed setup.py.

Thanks for pointing out this issue!

**Comment by ekaasine** [comment]
Hi Janne,

Thanks for the prompt reply. I downloaded the setup.py from the link and re-build MOODS but I get the same error?

**Comment by jhkorhonen** [comment]
Now that I test this myself, looks like setuptools does not re-build the extensions if just the setup.py changes. Can you still try cleaning up the previous build and then reinstalling, in case that helps:
```sh
python setup.py clean
rm MOODS/*.so
```
(If you did `python setup.py install`, also check that the old .so files in the library path get replaced.)

If that doesn't work, can you check if you can import the `MOODS.tools` and `MOODS.scan` modules? For example, does running `python ex-high-order-motif.py` crash also?

Also, what's your OS, C++ compiler and python version?

**Comment by ekaasine** [comment]
Yes, now I got it working after cleaning and removing everything and re-installing with the new setup.py. Thanks a lot!

**Comment by jhkorhonen** [comment]
Ok, great! I'll also publish a release with fixed `setup.py` soon™.

Thanks again for reporting the issue!

**Comment by jorgeboucas** [comment]
```bash
pip uninstall MOODS-python

rm -rf MOODS-python-1.9.1

wget https://github.com/jhkorhonen/MOODS/releases/download/v1.9.1/MOODS-python-1.9.1.tar.gz

tar -xjvf MOODS-python-1.9.1.tar.gz

cd MOODS-python-1.9.1

rm setup.py 

wget https://raw.githubusercontent.com/jhkorhonen/MOODS/develop/python/setup.py

python setup.py install --user

cd scripts

python moods_dna.py -m example-data/matrices/*.{pfm,adm} -s example-data/seq/chr1-5k-55k.fa -p 0.0001
```

working! Thanks!

👍

**Comment by jhkorhonen** [comment]
Fixed in [v1.9.2](https://github.com/jhkorhonen/MOODS/tree/v1.9.2). Installing MOODS 1.9.2 from the new release should also automatically overwrite the broken .so file in python library path.

**Comment by pwwang** [comment]
I totally followed @jorgeboucas's suggestions, but I still got:
```
Traceback (most recent call last):
  File "moods_dna.py", line 4, in <module>
    import MOODS.scan
  File "/home/m161047/.local/lib/python2.7/site-packages/MOODS/scan.py", line 28, in <module>
    _scan = swig_import_helper()
  File "/home/m161047/.local/lib/python2.7/site-packages/MOODS/scan.py", line 24, in swig_import_helper
    _mod = imp.load_module('_scan', fp, pathname, description)
ImportError: /home/m161047/.local/lib/python2.7/site-packages/MOODS/_scan.so: undefined symbol: _ZNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEE10_M_replaceEmmPKcm
```
Any ideas?
Thanks!
